recipes/omr1/pipeline/tir_solution_generation.py

```diff
@@ -0,0 +1,277 @@
+        f"--input_file={output_dir}/tir-preprocess-generations-{suffix}/preprocessed_output.jsonl "
+        f"--output_file={fragments_file} "
+        f"--window_size=1500"
+    )
+    
+    expname = f"{expname}-tir-extract-python-fragments-{suffix}"
+    run_cmd(
+        ctx=wrap_arguments(extraction_cmd),
+        cluster=cluster,
+        partition="cpu",
+        run_after=run_after,
+        log_dir=f"{output_dir}/tir-filter-novelty-significance-{suffix}/logs",
+        exclusive=True,
+        expname=expname,
+    )
+    run_after = expname
+
+    expname = f"{expname}-tir-judge-novelty-{suffix}"
+    generate(
+        ctx=wrap_arguments(
+            f"++prompt_template=qwen-instruct "
+            f"++prompt_config=/nemo_run/code/recipes/omr1/prompts/classify-tir-novelty.yaml "
+            f"++batch_size=512 "
+            f"++input_file={fragments_file} "
+            f"++generation_key=fragment_novelty "
+            f"++skip_filled=True "
+        ),
+        cluster=cluster,
+        output_dir=novelty_output_dir,
+        run_after=run_after,
+        model="/trt_models/qwen2.5-32b-instruct",
+        server_type="trtllm",
+        server_gpus=8,
+        server_nodes=1,
+        num_random_seeds=2,
+        # num_chunks=4,
+        expname=expname,
+    )
+    dependecies.append(expname)
+    
+    expname = f"{expname}-tir-judge-significance-{suffix}"
+    generate(
+        ctx=wrap_arguments(
+            f"++prompt_template=qwen-instruct "
+            f"++prompt_config=/nemo_run/code/recipes/omr1/prompts/classify-tir-significance.yaml "
+            f"++batch_size=512 "
+            f"++input_file={fragments_file} "
+            f"++generation_key=fragment_significance "
+            f"++skip_filled=True "
+        ),
+        cluster=cluster,
+        output_dir=significance_output_dir,
+        run_after=run_after,
+        model="/trt_models/qwen2.5-32b-instruct",
+        server_type="trtllm",
+        server_gpus=8,
+        server_nodes=1,
+        num_random_seeds=2,
+        # num_chunks=4,
+        expname=expname,
+    )
+    dependecies.append(expname)
+    
+    expname = f"{expname}-tir-filter-fragments-{suffix}"
+    run_cmd(
+        ctx=wrap_arguments(
+            f"python /nemo_run/code/recipes/omr1/scripts/filter_novelty_significance.py "
+            f"--novelty_files '{novelty_output_dir}/output-rs*.jsonl' "
+            f"--significance_files '{significance_output_dir}/output-rs*.jsonl' "
+            f"--output_file {output_dir}/tir-filter-novelty-significance-{suffix}/filtered_output.jsonl "
+        ),
+        cluster=cluster,
+        partition="cpu",
+        log_dir=f"{output_dir}/tir-filter-novelty-significance-{suffix}/logs",
+        exclusive=True,
+        run_after=dependecies, # run after novelty and significance judges
+        expname=expname,
+    )
+
+
+def prepare_for_sft(output_dir, suffix, cluster, expname, extra_args="", **generate_kwargs):
```

Comment:
if some functions are exactly the same as in other scripts, let's just import them?